Text suggestions for "The Whole Game" (#929)
hadley/r-pkgs: whole-game.Rmd

To keep the pace brisk, we exploit the modern conveniences in the devtools package and the RStudio IDE. In later chapters, we are more explicit about what those helpers are doing for us.

> We strongly suggest you follow along and create this toy package with us, however if you just want to read through it (or something goes awry halfway through!), completing it is not critical to understanding the remainder of the book.

Reviewer: Hmmm, I think this is close but not quite right. If this chapter is impenetrable, I think the rest of the book is going to be rough sailing indeed. I think we should only make a more limited promise that this chapter is self-contained.

Author: Right, that's a good point!

Author: Something like: "We strongly suggest you follow along and create this toy package with us, however it is self-contained and completing it is not a requirement to advancing through the remainder of the book."?

Author: Pushed an alternate wording that is hopefully better.

## Load devtools and friends

You can initiate your new package from any active R session.
You don't need to worry about whether you're in an existing or new project or not.

> The functions we use make sure that the package we will create is in a new, clean, project.

Reviewer: ```suggestion
The functions we use ensure that we create a new clean project for the package.
```

A select few are visible all the time, but sometimes you want to see them all.
:::

> -   `.Rbuildignore` lists files that we need to have around but that should not be included when building the R package from source. By default, this file (as well as `.gitignore`) is only created by `create_package()` if you are using RStudio, however you will likely need to make use of it in the future regardless of what editor you are using. It is discussed in more detail in @sec-rbuildignore.

Reviewer: I wasn't really conscious of this, but I did the experiment and see what you mean. Now that I get it, I'm going to take a crack at rewording.

```suggestion
-   `.Rbuildignore` lists files that we need to have around but that should not be included when building the R package from source. If you aren't using RStudio, `create_package()` may not create this file (nor `.gitignore`) at first, since there's no RStudio-related machinery that needs to be ignored. However, you will likely develop the need for `.Rbuildignore` at some point, regardless of what editor you are using. It is discussed in more detail in @sec-rbuildignore.
```

Author: I like that, it's clearer

Reviewer: Forgive me if you did this or already know this, but just in case, you can accept suggestions like this by clicking on "Commit suggestion" in this PR interface here in the browser. Then it will give you a chance to commit right away or build up a batch of suggestions and commit as one. Then you'd want to pull these new changes into your local branchsd right away.

Author: Yup, I know and love that feature!
